libminifi/src/provenance/ProvenanceRepository.cpp

```diff
@@ -36,6 +36,7 @@ void ProvenanceRepository::run() {
     uint64_t curTime = getTimeMillis();
     uint64_t size = repoSize();
     if (size >= purgeThreshold) {
+      std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
What makes this necessary? is there a way to avoid locking? LevelDB is thread safe due to process level locking.